operator-pipeline-images/operatorcert/entrypoints/apply_test_waivers.py

```diff
@@ -0,0 +1,124 @@
+"""Configure test suite for given operator by adding/removing labels"""
+import argparse
+import logging
+import os
+import re
+from typing import Any, Dict
+
+from github import Auth, Github
+from operatorcert.github import add_or_remove_labels
+from operatorcert.logger import setup_logger
+from operatorcert.utils import get_repo_config
+
+LOGGER = logging.getLogger("operator-cert")
+
+
+def setup_argparser() -> argparse.ArgumentParser:
+    """
+    Setup argument parser
+
+    Returns:
+        Any: Initialized argument parser
+    """
+    parser = argparse.ArgumentParser(
+        description="Github cli tool to manage required tests."
+    )
+    parser.add_argument(
+        "--github-host-url",
+        default="https://api.github.com",
+        help="The GitHub host, default: https://api.github.com",
+    )
+    parser.add_argument(
+        "--repo-config-file",
+        required=True,
+        help="A path to a repository configuration file.",
+    )
+
+    parser.add_argument(
+        "--operator-name",
+        required=True,
+        help="Name of the operator to be tested.",
+    )
+
+    parser.add_argument(
+        "--pull-request-url",
+        required=True,
+        help="The GitHub pull request URL where we want to edit labels.",
+    )
+
+    parser.add_argument("--verbose", action="store_true", help="Verbose output")
+    return parser
+
+
+def can_ignore_test(
+    test: Dict[str, Any],
+    operator_name: str,
+) -> bool:
+    """
+    Check if test can be ignored for given operator.
+
+    Args:
+        test (Dict[str, Any]): A test configuration to check.
+        operator_name (str): An operator name to check.
+
+    Returns:
+        bool: True if test can be ignored, False otherwise.
+    """
+    test_name = test.get("name")
+    LOGGER.debug("Checking test %s", test)
+
+    for ignored_operator in test.get("ignore_operators", []):
+        LOGGER.debug("Checking ignored operator %s", ignored_operator)
+        if not re.match(ignored_operator, operator_name):
```

Comment:
It might not matter in this case, but do we want to match only a substring of the test name here?

Reply:
Our team will define these regex exceptions so we have control over what's matched. Is there any concern that this implementation might not cover?

Reply:
You can see real config with regexes here: https://github.com/redhat-openshift-ecosystem/community-operators-prod/pull/3846/files